### PR TITLE
`react` and `graphql-tag` dependency adjustments

### PR DESCRIPTION
**Pull request by hwillson**
Includes the following:

- Makes `react` an optional dependency (since `@apollo/client`'s React integration can be removed using tree-shaking / dead code elimination)
- Updates React related dependencies
- Makes `graphql-tag` a full dependency, and re-exports `gql` making it part of the default `@apollo/client` bundle (to pave the way for future `gql` optimizations)
- Updates the docs to import `gql` from `@apollo/client`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -48,7 +48,7 @@
     {
       "name": "apollo-client",
       "path": "./dist/apollo-client.cjs.min.js",
-      "maxSize": "22.75 kB"
+      "maxSize": "33 kB"
```
Comment (benjamn) [inline package.json]: This seems like a very large jump, but I think I know how to fix it.

File: package.json [hunk 2]
```diff
@@ -52,14 +52,16 @@
     }
   ],
   "peerDependencies": {
-    "@types/react": "^16.8.0",
-    "graphql": "^14.5.4",
-    "react": "^16.8.0"
+    "graphql": "^14.5.4"
+  },
+  "optionalDependencies": {
+    "react": "^16.10.2"
```
Comment (SimenB) [inline package.json]: any chance of keeping this at `^16.8` to be compatible with the latest release of Expo (`16.8.3`)? RN is still on `16.9` as well

File: package.json (hunk 1 above)
Comment (hwillson) [reply package.json]: Thanks @benjamn - I was getting ahead of myself here, envisioning `gql` functionality being part of AC which is why I left it in the bundle. Thanks for fixing!

File: package.json (hunk 2 above)
Comment (hwillson) [reply package.json]: @SimenB - for sure, I'll get this adjusted separately. Thanks!

**Pull request merged**